File: java/MASTER_SLAVE_WHOLE_PART_PROXY/ProxyAutorizacionPago.java
public class ProxyAutorizacionPago extends SistemaAutorizacionPago{
        //Atributos
        Reservacion reservacion = new Reservacion();

        //Constructor
        public ProxyAutorizacionPago(){}

        //Métodos
        public void preparaCargo(DatosFuncion df){
            int cargo = 100;
            boolean reembolsable = false;
            super.cargoSolicitado.setCargo(cargo, reembolsable);
            super.solicitaAutorizacionPago(cargoSolicitado);
        }
}

File: python/MASTER_SLAVE_WHOLE_PART_PROXY/MASTER_SLAVE_WHOLE_PART_PROXY/proxy/ProxyAutorizacionPago.py
from ..whole_part.Reservacion import Reservacion
from .SistemaAutorizacionPago import SistemaAutorizacionPago
from .CargoSolicitado         import CargoSolicitado
        
class ProxyAutorizacionPago(SistemaAutorizacionPago):
    
    reservacion = Reservacion()
    
    def preparaCargo(self, datosFuncion, datosCompra):
        # Lógica que determina el cargo ...
        cargo = 100
        reembolsable = False
        self.cargoSolicitado = CargoSolicitado()
        self.cargoSolicitado.setCargo(cargo, reembolsable)
        self.solicitaAutorizacionCargo(self.cargoSolicitado)
        